tillerinobot/src/main/java/tillerino/tillerinobot/RecommendationsManager.java

```diff
@@ -343,122 +342,137 @@ private GivenRecommendation toGivenRecommendation(
 				System.currentTimeMillis(), sample.getMods());
 	}
 
-	public Settings parseSamplerSettings(OsuApiUser apiUser, @Nonnull String message,
-			Language lang) throws UserException, SQLException, IOException {
-		String[] remaining = message.split(" ");
-		
-		Settings settings = new Settings();
-		RecommendationType[] rt;
-		if(dataManager!=null)
-		{
-		   rt = dataManager.getData(apiUser.getUserId()).getRecommendMods();
+	public static Settings parseSamplerSettings(@CheckForNull BotBackend backend, UserDataManager dataManager, OsuApiUser apiUser, @Nonnull String message,
+		    Language lang) throws UserException, SQLException, IOException {
+
+	    	if(message.length() == 0)
+	    		message = dataManager.getData(apiUser.getUserId()).getRecommendationDefault().replaceAll("[,.]", "").replace("| ", "");
+		    String[] remaining = message.split(" ");
+		    Settings settings = new Settings();
+		    settings.model = Model.GAMMA;
+		    boolean nomodError = false;
+		    boolean modelError = false;
+		    
+		    
+		    for (int i = 0; i < remaining.length; i++) {
+
+		        String param = remaining[i];
+		        String lowerCase = param.toLowerCase();
+		        if (lowerCase.length() == 0) 
+		            continue;
+
+		        if (lowerCase.equals("any")) {
```

Comment:
no `getLevenshteinDistance`? :stuck_out_tongue: seems to be used on "nomod"/"relax" and stuff... idk xd